.github/workflows/deploy-mfe-s3.yml

```diff
@@ -0,0 +1,108 @@
+name: MFE S3 Bucket Deployment 🚀
+
+env:
+  RELEASE_VERSION: unset ## Set this variable in your branch to activate the production deployments.
+
+on:
+  push:
+    branches:
+      - unset ## Specify the branches that should trigger this action.
+
+  pull_request:
+    branches:
+       - "**open-rc**"
+jobs:
+  build:
+    environment:
+      name: ${{ github.ref_name == 'open-release/${{ env.RELEASE_VERSION }}.nelp' && 'prod' || 'stage' }}
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: "Echo job  vars"
+      env:
+        JOB_VARS: ${{ toJson(vars) }}
+      run: echo "$JOB_VARS"
+
+    - name: checkout mfe repo
+      uses: actions/checkout@v3
+
+    - name: Use Node.js ${{ vars.NODE_VERSION }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ vars.NODE_VERSION }}
+
+    - name: Cache node modules
+      id: cache-npm
+      uses: actions/cache@v3
+      env:
+        cache-name: cache-node-modules
+      with:
+        # npm cache files are stored in `~/.npm` on Linux/macOS
+        path: ~/.npm
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
+
+    - if: ${{ steps.cache-npm.outputs.cache-hit != 'true' }}
+      name: List the state of node modules
+      continue-on-error: true
+      run: npm list
+
+    - name: npm Install
+      run: npm install
+
+    - name: npm Build  # check env variables of repo.
+      run: npm run build
+      env:
+        PUBLIC_PATH: ${{ vars.PUBLIC_PATH_CDN }}
+        APP_ID: ${{ vars.APP_ID }}
+        MFE_CONFIG_API_URL: ${{ vars.MFE_CONFIG_API_URL }}
+        EXAMS_BASE_URL: ''
```

Comment:
I am trying to understand why this line is like not working. 
https://github.com/openedx/frontend-build/blob/v12.8.42/config/webpack.prod.config.js#L28-L30

Reply:
@andrey-canon Finally I think I catched. Everything is related to paragon alpha used in frontend build.

That line in frontend-built that use dotenv dont override by default. So every env var already configured would be keeped.
https://github.com/motdotla/dotenv?tab=readme-ov-file#override

The question know is who is setting to process.env the development vars...
Then I found that `webpack.common` is loaded upper.
https://github.com/nelc/frontend-build/blob/open-release/palm.nelp/config/webpack.prod.config.js#L23

And continuing the flow `webpack.common` with the alpha version there is an extra line only in the alpha version for paragon.

https://github.com/nelc/frontend-build/blob/open-release/palm.nelp/config/webpack.common.config.js#L4

That mysterious and hidden  paragon load file,  do the change.

https://github.com/nelc/frontend-build/blob/open-release/palm.nelp/lib/plugins/paragon-webpack-plugin/ParagonWebpackPlugin.js#L20-L22

Reply:
Thanks for your detailed answer, I have opened this [PR](https://github.com/nelc/frontend-app-learning/pull/27) based on your comments